### PR TITLE
doc: add a README to explain how the ransomware lua artifacts works

### PR DESCRIPTION
**Pull request by ayfaouzi**

### CODE REVIEW

**Comment by Mikaayenson** [comment]
@ayfaouzi Is there a repo to link issues to? See the [gh workflow](https://github.com/elastic/protections-artifacts/blob/2431aad4558ff59f9ea156225790c7800e344831/.github/workflows/duplicate_issue.yml).

**Review by Mikaayenson — APPROVED** [review]

File: .github/workflows/duplicate_issue.yml [hunk 1]
```diff
@@ -32,3 +32,16 @@ jobs:
...
+
+      - name: Duplicate issue to internal ransomware repo
+        uses: octokit/request-action@v2.x
+        if: contains(github.event.issue.labels.*.name, 'ransomware')
+        with:
+          route: POST /repos/{owner}/{repo}/issues
+          owner: elastic
+          repo: protections-ransomware
+          title: "${{ toJSON(format('{0} {1}', '[Openness Issue]', github.event.issue.title ))}}"
+          body: ${{toJSON(github.event.issue.body)}}
```
Comment (magermark) [inline .github/workflows/duplicate_issue.yml:44]: Could we modify the body to include a URI back to the original issue from the openness repo?

File: ransomware/README.md [hunk 2]
```diff
@@ -0,0 +1,26 @@
+# Elastic Security Ransomware Protection Artifact Code
+
+## How it works
+
+The `artifact.lua` contains the Elastic Security detection logic to detect ransomware. This file is loaded by the [endpoint](https://www.elastic.co/security/endpoint-security).
```
Comment (magermark) [inline ransomware/README.md:5]:
```suggestion
`artifact.lua` contains Elastic Security's ransomware detection logic. This file is loaded and operated by the [endpoint](https://www.elastic.co/security/endpoint-security).
```

File: ransomware/README.md [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+## Detection features
+
+- We use a wide variety of techniques to detect anomalous behavior:
+    - ✔️ **Header Mismatch**: when the file header magic does not match the expected sequence of bytes.
+    - ✔️ **Entropy Mismatch**: when the file entropy is larger than the expected entropy range.
+    - ✔️ **Path History**: evaluates the current event and how it relates to previous events involving the same filepath within the same process. In particular, we seek to find anomalous file modification patterns that may not be apparent when analyzing the current event in a vacuum (e.g. deleting and creating the same filepath).
+    - ✔️ **Abnormal Extensions**: performs various heuristics over the sub parts of a file extension.
+    - ✔️ **Ransom Notes**: checking for various pattern found in random notes dropped by ransomware.
+    - ✔️**Canary Files**: drop several canary files and monitoring them for encryption.
```
Comment (magermark) [inline ransomware/README.md:21]:
```suggestion
    - ✔️**Canary Files**: drop several canary files and monitor them for signs of encryption.
```

**Review by magermark — CHANGES_REQUESTED** [review]
Minor edit suggestions. I would also recommend updating the image to remove explicit references to `BigKahuna.sys` (since it is an internal name) as well as overt references to minifilter drivers and I/O completion ports. I would suggest the fs minifilter section to be replaced with a more abstract block that instead references Elastic's eventing system without getting into detail as to how the events are obtained.

File: .github/workflows/duplicate_issue.yml (hunk 1 above)
Comment (ayfaouzi) [reply .github/workflows/duplicate_issue.yml:44]: Yes we can `github.event.issue.html_url` contains the URL of the original issue:

Does this look ok ? 

```
body: "${{ toJSON(format('{0}\n\n{1}', github.event.issue.html_url,  github.event.issue.body ))}}"
```

**Review by magermark — APPROVED** [review]
LGTM!

**Pull request merged**